unified-codes/libs/ui/src/lib/components/entity/organisms/EntityBrowser.stories.tsx

```diff
@@ -1,32 +1,340 @@
+  ]);
+
+  return { types, setTypes };
+}
+
+export const withProps = () => {
+  const classes = useStyles();
+
+  // Table w/ pagination.
+
+  const {
+    entities,
+    count,
+    columns,
+    filterBy,
+    orderBy,
+    orderDesc,
+    setFilterBy,
+    setOrderDesc,
+  } = useTableState();
+
+  const {
+    page,
+    rowsPerPage,
+    rowsPerPageOptions,
+    setRowsPerPage,
+    setPage,
+  } = useTablePaginationState();
+
+  const data = React.useMemo(() => {
+    const start = rowsPerPage * page;
+    const end = rowsPerPage * (page + 1);
+    const filteredEntities = filterBy.length === 0 ? entities : entities.filter((entity: IEntity) => entity.description.includes(filterBy));
+    return filteredEntities.slice(start, end).sort((a: IEntity, b: IEntity) => {
+      if (a[orderBy] > b[orderBy]) return orderDesc ? -1 : 1;
+      if (b[orderBy] > a[orderBy]) return orderDesc ? 1 : -1;
+      return 0;
+    })
+   }, [entities, orderBy, orderDesc, rowsPerPage, page]);
+
+  const onSort = React.useCallback((_: string) => setOrderDesc(!orderDesc), [orderDesc, setOrderDesc]);
+
+  const onChangePage = React.useCallback((_: React.MouseEvent<HTMLButtonElement> | null, page: number) => 
+    setPage(page), [setPage]);
+
+  const onChangeRowsPerPage = React.useCallback((event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => 
+    setRowsPerPage(+event.target.value), [setRowsPerPage]);
+
+  const entityBrowserTableHeader = React.useMemo(() => (
+    <EntityTableHeader
+      classes={classes?.header}
+      columns={columns}
+      onSort={onSort}
+      orderDesc={orderDesc}
+      orderBy={orderBy}
+    />
+  ), [classes, columns, onSort, orderDesc, orderBy]);
+
+  const entityTableRows = React.useMemo(() => (
+    <React.Fragment>
+      {
+        data.map((entity: IEntity, index: number) => (
+          <EntityTableRow
+            classes={{
+              root: index % 2 ? classes?.table?.rowPrimary : classes?.table?.rowSecondary,
+              cell: classes?.table?.cell
+            }}
+            columns={columns}
+            entity={entity}
+            key={entity.code}
+          ></EntityTableRow>
+        ))
+      }
+    </React.Fragment>
+  ), [classes, columns, data]);
+    
+  const entityBrowserTable = React.useMemo(() => (
+    <EntityTable 
+      classes={classes?.table}
+      header={entityBrowserTableHeader}
+      rows={entityTableRows}
+    />
+  ), [classes, entityBrowserTableHeader, entityTableRows]);
+
+  const entityBrowserTablePagination = React.useMemo(() => (
+    <TablePagination
+      classes={{ root: classes?.tablePagination?.root }}
+      rowsPerPageOptions={rowsPerPageOptions}
+      component="div"
+      count={count}
+      rowsPerPage={rowsPerPage}
+      page={page}
+      onChangePage={onChangePage}
+      onChangeRowsPerPage={onChangeRowsPerPage}
+    />
+  ), [classes, rowsPerPageOptions, count, rowsPerPage, page, onChangePage, onChangeRowsPerPage]);
+
+  // Search bar.
```

Comment:
Not too much left to do re. `SearchBar`! Add classes for `buttonContainer` and `inputContainer`, pass up `handleKeyDown` as prop, and I think we're done!